op_crates/fetch/lib.rs

```diff
@@ -133,28 +141,79 @@ where
     return Err(type_error(format!("scheme '{}' not supported", scheme)));
   }
 
-  {
-    let state_ = state.borrow();
-    let permissions = state_.borrow::<FP>();
-    permissions.check_net_url(&url_)?;
-  }
+  let permissions = state.borrow::<FP>();
+  permissions.check_net_url(&url_)?;
 
   let mut request = client.request(method, url_);
 
-  match data.len() {
-    0 => {}
-    1 => request = request.body(Vec::from(&*data[0])),
-    _ => panic!("Invalid number of arguments"),
-  }
+  let maybe_request_body_rid = if args.has_body {
+    match data.len() {
+      0 => {
```

Comment:
This is an optimization for the majority of calls to fetch, where the request body is not streaming, and instead is synchronously available. In these cases the request body does not need to be passed to rust in an extra `op_fetch_request_write` and no `FetchRequestBodyResouce` needs to be created. Instead the request body is passed synchronously into Rust, like it was previously.